yaegi test golang.org/x/crypto/bcrypt fails at parse

### The following program `sample.go` triggers an unexpected result

```go
$ yaegi test -v golang.org/x/crypto/bcrypt
```


### Expected result

```console
$ yaegi test -v golang.org/x/crypto/bcrypt
== RUN   TestCost
-- PASS: TestCost (0.00s)
[...]
```


### Got

```console
$ yaegi test -v golang.org/x/crypto/bcrypt
test: /home/marc/go/src/golang.org/x/crypto/bcrypt/bcrypt.go:135:12: cannot use type untyped rune as type uint8 in assignment
```


### Yaegi Version

master (after #1231)

### Additional Notes

This was working (minus errors for some tests) in v0.9.23.

--- a/_test/rune2.go
+++ b/_test/rune2.go
@@ -1,0 +1,22 @@
+package main
+
+import "fmt"
+
+const majorVersion = '2'
+
+type hashed struct {
+	major byte
+}
+
+func main() {
+	fmt.Println(majorVersion)
+
+	p := new(hashed)
+	p.major = majorVersion
+
+	fmt.Println(p)
+}
+
+// Output:
+// 50
+// &{50}

--- a/interp/type.go
+++ b/interp/type.go
@@ -1773,15 +1773,19 @@ func (t *itype) defaultType(v reflect.Value, sc *scope) *itype {
 
 	typ := t
 	// The default type can also be derived from a constant value.
-	if v.IsValid() && t.TypeOf().Implements(constVal) {
-		// TODO: find a way to get actual types here
+	if v.IsValid() && v.Type().Implements(constVal) {
 		switch v.Interface().(constant.Value).Kind() {
 		case constant.String:
 			typ = sc.getType("string")
 		case constant.Bool:
 			typ = sc.getType("bool")
 		case constant.Int:
-			typ = sc.getType("int")
+			switch t.cat {
+			case int32T:
+				typ = sc.getType("int32")
+			default:
+				typ = sc.getType("int")
+			}
 		case constant.Float:
 			typ = sc.getType("float64")
 		case constant.Complex:

